ShoppingM8-be/src/main/java/edu/pw/shoppingm8/list/api/ListController.java

```diff
@@ -0,0 +1,104 @@
+package edu.pw.shoppingm8.list.api;
+
+import java.net.URI;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import edu.pw.shoppingm8.authentication.AuthenticationService;
+import edu.pw.shoppingm8.list.List;
+import edu.pw.shoppingm8.list.ListService;
+import edu.pw.shoppingm8.list.api.dto.ListDto;
+import edu.pw.shoppingm8.user.User;
+import io.swagger.annotations.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+
+@RestController
+@RequestMapping("list")
+@RequiredArgsConstructor
+@Slf4j
+public class ListController {
+    private final ListService listService;
+    private final AuthenticationService authenticationService;
+
+    @ApiOperation(value = "Create list", nickname = "create list", notes = "",
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "If list was created", response = ListDto.class),
+        @ApiResponse(code = 400, message = "If request has list id or owner id set")})
+    @PostMapping(value="/new")
+    ResponseEntity<ListDto> createList(@RequestBody ListDto listDto) {
+        if (listDto.getId() != null || listDto.getOwnerId() != null)
```

Comment:
This validation can be done by annotations on dto class (look up @NotNull and @Valid). then the controller layer will be simplified

Reply:
See that here it assumed that those fields must be null, not that they are not null

Reply:
Added separate DTO for list modification to improve code readability